### PR TITLE
Add Timolia

### PR DESCRIPTION
**Pull request by l3nnartt**
Add Timolia.de

### CODE REVIEW

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix all errors, upload proof of it working

**Comment by l3nnartt** [comment]
Screenshot 2021-08-07 000937
Screenshot 2021-08-07 001014
Screenshot 2021-08-07 001057
Screenshot 2021-08-07 001124

File: websites/T/timolia/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Timolia",
+    "author": {
+      "name": "Lennart™",
+      "id": "0001"
```
Comment (Bas950) [inline websites/T/timolia/dist/metadata.json:6]: Id should be your Discord ID, not your discriminator. 
Example (mine): 241278257335500811

File: websites/T/timolia/presence.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+const presence = new Presence({
+    clientId: "872421983554502717"
+  }),
+
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:4]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "large_image",
+    details: document.location.href,
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:10]: details to href????

File: websites/T/timolia/presence.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    startTimestamp: browsingStamp
+  };
+
+  if (document.location.hostname === "www.timolia.de") {
+    if (document.location.pathname === "/") {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:16]:
```suggestion
```
unneed, set at line 11

File: websites/T/timolia/presence.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        const player = document.querySelector("#liveplayercount").textContent;
+        presenceData.details = "Viewing home page";
+        presenceData.smallImageKey = "minecraft";
+        presenceData.smallImageText = `${player} online players`;
+    } else if (document.location.pathname.startsWith("/stats")) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:22]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        const stats = document.querySelector("#playername").textContent;
+        presenceData.details = "Viewing stats from:";
+        presenceData.state = stats;
+    } else if (document.location.pathname.includes("/games")) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:27]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        presenceData.details = "Viewing gamemodes";
+    } else if (document.location.pathname.includes("/account")) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:30]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        presenceData.details = "Viewing account settings";
+    }
+  } else if (document.location.hostname === "howto.timolia.de") {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:34]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        presenceData.details = "HowTo - Reading:";
+        presenceData.state = document.querySelector("body > div.md-container > main > div > div.md-content > article > h1").textContent;
+    } else if (document.location.hostname === "shop.timolia.de") {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:38]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        presenceData.details = "Shop - Viewing store page";
+        if (document.location.pathname.startsWith("/checkout")) {
+            presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:41]:
```suggestion
```

File: websites/T/timolia/presence.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+            presenceData.details = "Shop - Viewing shopping cart:";
+            presenceData.state = document.querySelector("body > div.sections-wrapper > section > div > div > div.jumbotron > div > div.packages > form > table > tbody > tr > td.col-md-6").textContent;
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:43]: What exactly would it show here?

File: websites/T/timolia/presence.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+            presenceData.state = forum;
+        }
+    } else if (document.location.hostname.includes("wi.timolia.de"))
+        presenceData.details = "Viewing Webinterface";
+    else if (document.location.hostname.includes("gitlab.timolia.de")) {
+        presenceData.details = "Viewing GitLab Server";
+        presenceData.smallImageKey = "gitlab";
+    }
+
+    if (presenceData.details === null) {
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:67]:
```suggestion
    if (!presenceData.details) {
```

File: websites/T/timolia/presence.ts [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+        presence.setTrayTitle();
+        presence.setActivity();
+    } else 
+        presence.setActivity(presenceData);
+
```
Comment (Bas950) [inline websites/T/timolia/presence.ts:72]:
```suggestion
    } else presence.setActivity(presenceData);
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/T/timolia/presence.ts (hunk 3 above)
Comment (l3nnartt) [reply websites/T/timolia/presence.ts:9]: href displays the current website url

File: websites/T/timolia/presence.ts (hunk 11 above)
Comment (l3nnartt) [reply websites/T/timolia/presence.ts:38]: The article you are looking at

Comment (l3nnartt) [reply websites/T/timolia/presence.ts:35]: Screenshot 2021-08-10 000212

File: websites/T/timolia/presence.ts (hunk 3 above)
Comment (Slowlife01) [reply websites/T/timolia/presence.ts:9]: *But why use it on details*

Comment (l3nnartt) [reply websites/T/timolia/presence.ts:9]: If you are on a website where there is no explicit description only the URL is shown

Comment (Bas950) [reply websites/T/timolia/presence.ts:9]: Then something like "Unsupported Page" or just nothing is better in my opinion :/

Comment (l3nnartt) [reply websites/T/timolia/presence.ts:9]: alright, then i remove it :)

**Review by Bas950 — CHANGES_REQUESTED** [review]
I guess update your fork or make sure these changes are gone 
image

**Comment by l3nnartt** [comment]
oh sorry, i reverted the changes!

**Review by Bas950 — APPROVED** [review]

File: websites/T/timolia/dist/metadata.json [hunk 14]
```diff
@@ -0,0 +1,25 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Timolia",
```
Comment (Bas950) [inline websites/T/timolia/dist/metadata.json:3]: Service name and folder name must be the same (caps included)

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/T/timolia/dist/metadata.json (hunk 14 above)
Comment (l3nnartt) [reply websites/T/timolia/dist/metadata.json]: fixed

**Review by l3nnartt — COMMENTED** [review]
fixed service name

**Review by Bas950 — COMMENTED** [review]
Shouldn't the T be in uppercase?

**Review by Bas950 — COMMENTED** [review]
folder name didnt update...

**Comment by l3nnartt** [comment]
i guess its fixed

**Review by Bas950 — CHANGES_REQUESTED** [review]
Metadata in wrong one

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**